### PR TITLE
fix: DnsHole controls timer

### PR DESCRIPTION
**Pull request by hillaliy**
<br/>
<div align="center">
  
  <h3>Homarr</h3>
</div>

**Thank you for your contribution. Please ensure that your pull request meets the following pull request:**

- [x] Builds without warnings or errors (``pnpm buid``, autofix with ``pnpm format:fix``)
- [x] Pull request targets ``dev`` branch
- [x] Commits follow the [conventional commits guideline](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] No shorthand variable names are used (eg. ``x``, ``y``, ``i`` or any abbrevation)

### CODE REVIEW

File: packages/widgets/src/dns-hole/controls/TimerModal.tsx
Comment (manuel-rw) [inline packages/widgets/src/dns-hole/controls/TimerModal.tsx:11]: This is the singular form. However, this is an array (hence pluaral should be used).
Also, ``integration`` sounds like an object. Please name it ``integrationIds`` since it passes the IDs

File: packages/widgets/src/dns-hole/controls/component.tsx
Comment (manuel-rw) [inline packages/widgets/src/dns-hole/controls/component.tsx:96]: Again, please use plural

File: packages/widgets/src/dns-hole/controls/component.tsx
Comment (manuel-rw) [inline packages/widgets/src/dns-hole/controls/component.tsx:179]: Use plural

**Review by manuel-rw — CHANGES_REQUESTED** [review]

**Comment by hillaliy** [comment]
I think it's can be merge

**Review by manuel-rw — APPROVED** [review]
LGTM. Please **wait for a second approval**

File: packages/integrations/src/adguard-home/adguard-home-integration.ts
Comment (SeDemal) [inline packages/integrations/src/adguard-home/adguard-home-integration.ts:127]: Small possible improvement, you can just set the default value of duration right in the function's parameter definition. Instead of "(duration?: number)", you can do "(duration = 0)".
You then won't need to test for undefined.
And instead of doing the *1000 here, apply it down in the body of the request.
(0*1000 is still 0 so it's fine)

File: packages/widgets/src/dns-hole/controls/component.tsx
Comment (SeDemal) [inline packages/widgets/src/dns-hole/controls/component.tsx:96]: We shouldn't disable integrations that are already disabled.
Oldmarr has a neet function that filters the integrations between the 2 subset.

Also, you did something that was quite nice in oldmarr, instead of setting the list of integrations, you just by default assumed it was all disabled integrations if a specific integration ID wasn't given. But that's not important, the first part matters more.

File: packages/widgets/src/dns-hole/controls/component.tsx
Comment (SeDemal) [reply packages/widgets/src/dns-hole/controls/component.tsx:184]: I unmarked the "resolve".
Either make the function accept only single element instead of an array, or make it plural, just because you use it with a single item doesn't make it singular. you still pass an array to it and It could be used as is otherwise.

Like I mentioned before though, if you made the state for single items to begin with and assume all integration when that state is unset, then you can set all of them to single and won't need to do specific things as such.

**Review by SeDemal — DISMISSED** [review]

File: packages/widgets/src/dns-hole/controls/component.tsx
Comment (Meierschlumpf) [inline packages/widgets/src/dns-hole/controls/component.tsx:126]: This should be a jsx component and not a function call

**Review by Meierschlumpf — CHANGES_REQUESTED** [review]

**Review by Meierschlumpf — APPROVED** [review]
Now look good to me

**Pull request merged**
